CorsixTH/Lua/entities/humanoids/patient.lua

```diff
@@ -295,12 +296,20 @@ function Patient:treatDisease()
   end
 end
 
+--! Sets a permanent price of treatment for this patient instance,
+--! so changing price in casebook wont affect any longer
+function Patient:setTreatmentPrice(disease_id)
+  local hosp = self.hospital
+  self.pay_amount = hosp:getTreatmentPrice(disease_id)
+end
+
 --! Returns true if patient agrees to pay for the given treatment.
 --!param disease_id (string): The id of the disease to test
 function Patient:agreesToPay(disease_id)
   local casebook = self.hospital.disease_casebook[disease_id]
   local price_distortion = self:getPriceDistortion(casebook)
   local is_over_priced = price_distortion > self.hospital.over_priced_threshold
+  self:setTreatmentPrice(disease_id)
```

Comment:
If you provide the price here, you don't need to look up the price through the hospital in 302/303.
That eliminates the casebook there completely.

I also wonder if you should store the disease alongside the amount in the patient, so you can check the patient is requested money for the same disease as was decided here.

Reply:
> If you provide the price here, you don't need to look up the price through the hospital in 302/303. That eliminates the casebook there completely.

So setTreatmentPrice() should be removed and code should move to agreesToPay?